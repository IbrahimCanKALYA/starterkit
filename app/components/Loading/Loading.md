# Loading

## Description

- Simple loading component that use native loading components for both side.

## Funtions

- absent !

## Props

- absent !