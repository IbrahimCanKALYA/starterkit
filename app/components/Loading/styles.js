/* @flow */
const styles = {
  container: {
    zIndex: 99999,
    backgroundColor: '#00000050',
    alignSelf: 'stretch',
    position: 'absolute',
    justifyContent: 'center',
    left: 0,
    top: 0,
    right: 0,
    bottom: 0,
  },
};

export default styles;
