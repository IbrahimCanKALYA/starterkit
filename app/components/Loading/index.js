/* @flow */
import React, { Component } from 'react';
import { View, ActivityIndicator } from 'react-native';
import styles from './styles';

type Props = any;

class Loading extends Component<Props> {
  static defaultProps: any;

  render() {
    return (
      <View style={styles.container}>
        <ActivityIndicator size="large" />
      </View>
    );
  }
}

Loading.propTypes = {};
Loading.defaultProps = {};

export default Loading;
