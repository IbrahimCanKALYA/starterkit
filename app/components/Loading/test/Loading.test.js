import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../index';

test('Loading renders correctly', () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
