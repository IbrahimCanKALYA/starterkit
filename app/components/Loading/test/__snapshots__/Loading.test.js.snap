// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#00000050",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 99999,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
